src/linearmixedmodel.jl

```diff
@@ -782,6 +782,45 @@ StatsBase.residuals(m::LinearMixedModel) = response(m) .- fitted(m)
 
 StatsBase.response(m::LinearMixedModel) = m.y
 
+"""
+    restoreoptsum!(m::LinearMixedModel, io::IO)
+    restoreoptsum!(m::LinearMixedModel, fnm::AbstractString)
+
+Read, check, and restore the `optsum` field from a JSON stream or filename.
+"""
+function restoreoptsum!(m::LinearMixedModel, io::IO)
+    dict = JSON3.read(io)
+    ops = m.optsum
+    okay = (setdiff(propertynames(ops), keys(dict)) == [:lowerbd]) &&
+        isapprox(ops.initial, copy(dict.initial)) &&
+        isapprox(ops.xtol_abs, copy(dict.xtol_abs)) &&
+        ops.ftol_rel == dict.ftol_rel &&
+        ops.xtol_rel == dict.xtol_rel &&
+        ops.ftol_abs == dict.ftol_abs &&
+        ops.maxfeval == dict.maxfeval &&
```

Comment:
I'm wondering if we need this strict criterion. What happens if a user creates and fits a model, save its optsum, then modifies these parameters and then decides "nope, let's go back to the original fit"? 

Of course, not checking this would result in "clobbering" behavior.

Reply:
On thinking this over I favor the "clobbering" behavior.  The assumption is that the saved values came from a bona fide optimization and it makes sense to then restore them all.  The values in the model before being fit are the defaults and not necessarily meaningful.

I will add code to check that `objective(updateL!(setθ!(m, dict.final)))` is approximately `dict.fmin` and also compare `dict.finitial` to the evaluation at `dict.initial`.  (I will loosen the type signature of the second argument to `setθ!` to allow for this.)  We can't guarantee that we have the same model fit until we preserve the formula and contrasts but at least this will mean that the model fits are consistent.